// @migros/mdx-web-components/dist/esm/mdx-chip-group.entry.js
import { r as registerInstance, c as createEvent, h, H as Host, g as getElement } from './index-1942bff0.js';

const chipGroupCss = ":host{box-sizing:border-box;display:flex;flex-direction:row;flex-wrap:wrap;gap:var(--mdx-sys-spacing-fix-s)}";
const MdxChipGroupStyle0 = chipGroupCss;

const ChipContainer = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.chipChange = createEvent(this, "chipChange", 7);
        this.selectedChips = [];
        this.multiSelect = false;
    }
    toggle({ detail }) {
        const selectedChip = detail.selected ? { selected: detail.selected, chipId: detail.chipId } : undefined;
        if (this.multiSelect) {
            if (selectedChip !== undefined) {
                this.selectedChips.push(selectedChip);
            }
            else {
                this.selectedChips = this.selectedChips.filter((chip) => chip.chipId !== detail.chipId);
            }
            this.chipChange.emit(this.selectedChips);
        }
        else {
            this.chips.forEach((chip) => {
                if (detail.chipId !== chip.chipId && detail.selected) {
                    chip.selected = false;
                }
            });
            if (selectedChip !== undefined) {
                this.selectedChips = [selectedChip];
                this.chipChange.emit(this.selectedChips);
            }
            else if (this.chips.every((chip) => !chip.selected)) {
                this.selectedChips = [];
                this.chipChange.emit(this.selectedChips);
            }
        }
    }
    get chips() {
        return Array.from(this.host.querySelectorAll('mdx-chip'));
    }
    componentWillLoad() {
        this.chips.forEach((chip) => {
            if (!this.multiSelect) {
                chip.singleSelect = true;
            }
        });
    }
    render() {
        return (h(Host, { key: 'bdf03c778feccff3e5b3d065ca04ede91a900446' }, h("slot", { key: 'f43d21f90ae253dc6667ed2a4c477e812ee32f2e' })));
    }
    get host() { return getElement(this); }
};
ChipContainer.style = MdxChipGroupStyle0;

export { ChipContainer as mdx_chip_group };
