// @migros/mdx-web-components/dist/esm/mdx-breadcrumb.entry.js
import { r as registerInstance, h } from './index-399d9287.js';
import { I as ICON_SIZES } from './exports-f063c20a.js';

var BREADCRUMB_VARIANTS;
(function (BREADCRUMB_VARIANTS) {
  BREADCRUMB_VARIANTS["DEFAULT"] = "default";
  BREADCRUMB_VARIANTS["CURRENT"] = "current";
})(BREADCRUMB_VARIANTS || (BREADCRUMB_VARIANTS = {}));

const breadcrumbCss = ":host{display:flex;align-items:center}:host .breadcrumb{display:inline-flex;align-items:center;color:var(--mdx-comp-breadcrumbs-color-default);font:var(--mdx-comp-breadcrumbs-font-default);letter-spacing:var(--mdx-comp-breadcrumbs-font-default-letter-spacing);text-transform:var(--mdx-comp-breadcrumbs-font-default-text-case);text-decoration:var(--mdx-comp-breadcrumbs-font-default-text-decoration);cursor:pointer}:host .breadcrumb ::slotted(*){max-width:100px;overflow:hidden;white-space:nowrap;text-overflow:ellipsis}:host .breadcrumb:hover{color:var(--mdx-comp-breadcrumbs-color-hover)}:host .breadcrumb__chevron-right{margin-left:var(--mdx-comp-breadcrumbs-spacing-gap-label-chevron);color:var(--mdx-comp-breadcrumbs-icon-color-chevron)}:host([variant=current]) .breadcrumb{color:var(--mdx-comp-breadcrumbs-color-active);cursor:default}:host([variant=current]) .breadcrumb ::slotted(*){max-width:none}";

const Breadcrumb = class {
  constructor(hostRef) {
    registerInstance(this, hostRef);
    this.disabled = false;
    this.variant = BREADCRUMB_VARIANTS.DEFAULT;
    this.href = undefined;
    this.ariaLabel = undefined;
    this.target = '_self';
  }
  render() {
    return (h("a", { class: "breadcrumb", href: this.href, target: this.target, "aria-label": this.ariaLabel, title: this.ariaLabel }, h("slot", null), this.variant === BREADCRUMB_VARIANTS.DEFAULT ? (h("mdx-icon-chevron-right", { class: "breadcrumb__chevron-right", size: ICON_SIZES.XS })) : null));
  }
};
Breadcrumb.style = breadcrumbCss;

export { Breadcrumb as mdx_breadcrumb };
