// @migros/mdx-web-components/dist/esm/mdx-breadcrumb.entry.js
import { r as registerInstance, h } from './index-7e8279cb.js';
import { I as ICON_SIZES } from './exports-6dd4526a.js';

const BREADCRUMB_VARIANTS = {
    DEFAULT: 'default',
    CURRENT: 'current',
};

const breadcrumbCss = ":host{display:flex;align-items:center}:host .breadcrumb{display:inline-flex;align-items:center;color:var(--mdx-comp-breadcrumbs-color-default);font:var(--mdx-comp-breadcrumbs-font-default);letter-spacing:var(--mdx-comp-breadcrumbs-font-default-letter-spacing);text-transform:var(--mdx-comp-breadcrumbs-font-default-text-case);text-decoration:var(--mdx-comp-breadcrumbs-font-default-text-decoration);cursor:pointer}:host .breadcrumb:focus-visible{border-radius:var(--mdx-comp-keyboard-focus-border-radius-default);outline-color:var(--mdx-comp-keyboard-focus-color-default);outline-style:solid;outline-width:var(--mdx-comp-keyboard-focus-border-width-default);outline-offset:2px}:host .breadcrumb ::slotted(*){max-width:100px;overflow:hidden;white-space:nowrap;text-overflow:ellipsis}:host .breadcrumb:hover{color:var(--mdx-comp-breadcrumbs-color-hover)}:host .breadcrumb__chevron-right{margin-left:var(--mdx-comp-breadcrumbs-spacing-gap-label-chevron);color:var(--mdx-comp-breadcrumbs-icon-color-chevron)}:host([variant=current]) .breadcrumb{color:var(--mdx-comp-breadcrumbs-color-active);cursor:default}:host([variant=current]) .breadcrumb ::slotted(*){max-width:none}";
const MdxBreadcrumbStyle0 = breadcrumbCss;

const Breadcrumb = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.disabled = false;
        this.variant = BREADCRUMB_VARIANTS.DEFAULT;
        this.href = undefined;
        this.ariaLabel = undefined;
        this.target = '_self';
    }
    render() {
        return (h("a", { key: '8a84260d0f71c3325d274d7c04dc222ab89bf322', class: "breadcrumb", href: this.href, target: this.target, "aria-label": this.ariaLabel, title: this.ariaLabel }, h("slot", { key: 'bf1111de3894e39a71c04b9392ff5578a814b8a5' }), this.variant === BREADCRUMB_VARIANTS.DEFAULT ? (h("mdx-icon-chevron-right", { class: "breadcrumb__chevron-right", size: ICON_SIZES.XS })) : null));
    }
};
Breadcrumb.style = MdxBreadcrumbStyle0;

export { Breadcrumb as mdx_breadcrumb };
